doc/rtd/explanation/format.rst

```diff
@@ -141,66 +225,128 @@ Create user data containing 3 shell scripts:
-`boothook` must take care of this itself.
+Explanation
+-----------
 
-It is provided with the instance id in the environment variable
-``INSTANCE_ID``. This could be made use of to provide a 'once-per-instance'
-type of functionality.
+`Jinja templating <https://jinja.palletsprojects.com/>`_ may be used for
+cloud-config and user data scripts. Any
+:ref:`instance-data variables<instance_metadata-keys>` may be used
+as jinja template variables. Any jinja templated configuration must contain
+the original header along with the new jinja header above it.
 
-Begins with: ``#cloud-boothook`` or ``Content-Type: text/cloud-boothook`` when
-using a MIME archive.
+.. note::
+    Use of Jinja templates is ONLY supported for cloud-config and user data
+    scripts. Jinja templates are not supported for cloud-boothooks or
+    meta configs.
 
-Part-handler
+Include file
 ============
 
-This is a `part-handler`: It contains custom code for either supporting new
-mime-types in multi-part user data, or overriding the existing handlers for
-supported mime-types.  It will be written to a file in
-:file:`/var/lib/cloud/data` based on its filename (which is generated).
+| **Header:** #include
+| **Content-Type:** text/x-include-url
 
-This must be Python code that contains a ``list_types`` function and a
-``handle_part`` function. Once the section is read the ``list_types`` method
-will be called. It must return a list of mime-types that this `part-handler`
-handles. Since MIME parts are processed in order, a `part-handler` part
-must precede any parts with mime-types it is expected to handle in the same
-user data.
+Example
+-------
 
-The ``handle_part`` function must be defined like:
+.. code-block:: text
 
-.. code-block:: python
+    #include
+    https://raw.githubusercontent.com/canonical/cloud-init/403f70b930e3ce0f05b9b6f0e1a38d383d058b53/doc/examples/cloud-config-run-cmds.txt
+    https://raw.githubusercontent.com/canonical/cloud-init/403f70b930e3ce0f05b9b6f0e1a38d383d058b53/doc/examples/cloud-config-boot-cmds.txt
 
-    def handle_part(data, ctype, filename, payload):
-      # data = the cloudinit object
-      # ctype = "__begin__", "__end__", or the mime-type of the part that is being handled.
-      # filename = the filename of the part (or a generated filename if none is present in mime data)
-      # payload = the parts' content
+Explanation
+-----------
 
-``Cloud-init`` will then call the ``handle_part`` function once before it
-handles any parts, once per part received, and once after all parts have been
-handled. The ``'__begin__'`` and ``'__end__'`` sentinels allow the part
-handler to do initialisation or teardown before or after receiving any parts.
+An include file contains a list of URLs, one per line. Each of the URLs will
+be read and their content can be any kind of user data format, both base
+config and meta config. If an error occurs reading a file the remaining files
+will not be read.
+
+Gzip compressed content
```

Comment:
Is this format supported on all platforms? I seem to recall there being some platform-specific oddities about this but don't remember specifics.

Reply:
As far as I know it should. If your cloud mangles your user data (like LXD does), it won't work, but that's the only limitation I'm aware of.